contributing/contribution_guide.rst

```diff
@@ -0,0 +1,128 @@
+
+.. code-block:: php
+
+    $ cd pim-community-dev
+    $ git remote add upstream git://github.com/akeneo/pim-community-dev.git
+
+Create a new branch
+
+.. code-block:: php
+
+    $ git checkout -b BRANCH_NAME master
+
+.. note::
+
+    How to name your branch ?
+
+Now, in order to install Akeneo PIM follow this guide http://docs.akeneo.com/latest/developer_guide/installation/installation_workstation.html.
+
+Work on your branch
+-------------------
+
+Before working on a contribution for an Akeneo repository, please read the following code convention
+http://docs.akeneo.com/latest/reference/best_practices/core/conventions.html and coding standard
+http://docs.akeneo.com/latest/reference/best_practices/core/standards.html to be sure to respect all our standard
+and avoid a lot of comments from core developers on your final Pull Request.
+
+For all contributions, please keep in mind that tests are as important as business code.
+Behavior of the application has to be tested with Behat.
+
+.. note::
+
+    How to setup Behat in Akeneo PIM http://docs.akeneo.com/latest/reference/best_practices/core/behat.html
+    How to begin with Behat http://docs.behat.org/en/v2.5/quick_intro.html
+
+Behavior of the business code has to be tested with PHPSpec.
+
+.. note::
+
+    How to begin with PHPSpec http://www.phpspec.net/en/latest/ and Prophecy https://github.com/phpspec/prophecy#prophecy
+
+For a bug fix contribution, please avoid any BC breaks. If a BC break can't be avoided add a comment and detail why.
+
+Commit your code
+----------------
+
+Begin by adding file content to your index
+
+.. code-block:: php
+
+    $ git add -p
+
+This will run a `git add` command with an interactive mode. You'll be able to choose which chunk of code you want to add.
+
+Create atomic and logical commits with a relevant message.
+
+It will help us to :
+ - Speed up the reviewing process
+ - Revert a single commit if needed
+ - Cherry pick a commit if needed
+
+Example of a well formed commit message (from github doc https://git-scm.com/book/ch5-2.html)
+
+.. note::
+
+    Short (50 chars or less) summary of changes
+
+    More detailed explanatory text, if necessary.  Wrap it to
+    about 72 characters or so.  In some contexts, the first
+    line is treated as the subject of an email and the rest of
+    the text as the body.  The blank line separating the
+    summary from the body is critical (unless you omit the body
+    entirely); tools like rebase can get confused if you run
+    the two together.
+
+    Further paragraphs come after blank lines.
+
+      - Bullet points are okay, too
+
+      - Typically a hyphen or asterisk is used for the bullet,
+        preceded by a single space, with blank lines in
+        between, but conventions vary here
+
+
+.. code-block:: php
+
+    $ git commit
+
+Submitting a Pull Request
+-------------------------
+
+Simple rules to follow before submitting a PR :
+ - Squash irrelevant commits
```

Comment:
[Github supports squash's](https://github.com/blog/2141-squash-your-commits). May be it is not so actual now.